linetools/isgm/abscomponent.py

```diff
@@ -594,6 +598,49 @@ def repr_alis(self, T_kin=1e4*u.K, bturb=0.*u.km/u.s,
         s += '\n'
         return s
 
+    def repr_joebvp(self, specfile, flags=(2,2,2), b_default=10*u.km/u.s):
+        """
+        String representation for JOEBVP (line fitting software).
+
+        Parameters
+        ----------
+        specfile : str
+            Name of the spectrum file
+        flags : tuple of ints, optional
+            Flags (nflag, bflag, vflag). See JOEBVP input for details
+            about these flags.
+        b_default : Quantity, optional
+            Doppler parameter value adopted in case an absorption
+            line within the component has not set this attribute
+            Default is 10 km/s.
+
+        Returns
+        -------
+        repr_joebvp : str
+            May contain multiple "\n" (1 per absline within component)
+
+        """
+        # Reference:
+        # specfile|restwave|zsys|col|bval|vel|nflag|bflag|vflag|vlim1|vlim2|wobs1|wobs2|trans
+        s = ''
+        for aline in self._abslines:
+            s += '{:s}|{}|'.format(specfile, aline.wrest.to('AA').value)
+            logN = aline.attrib['logN']
+            b_val = aline.attrib['b'].to('km/s').value
+            if b_val == 0:  # set the default
+                b_val = b_default.to('km/s').value
+            s += '{}|{}|{}|0.|'.format(self.zcomp, logN, b_val)  # `vel` is set to 0. because z is zcomp
+            s += '{}|{}|{}|'.format(flags[0], flags[1], flags[2])
```

Comment:
Use explicit formatting here to avoid the floating point issues in Python 3,e.g.

{:0.4f} instead of {}